src/unix/core.c

```diff
@@ -1197,7 +1198,8 @@ int uv__getpwuid_r(uv_passwd_t* pwd) {
   name_size = strlen(pw.pw_name) + 1;
   homedir_size = strlen(pw.pw_dir) + 1;
   shell_size = strlen(pw.pw_shell) + 1;
-  pwd->username = uv__malloc(name_size + homedir_size + shell_size);
+  gecos_size = strlen(pw.pw_gecos) + 1;
+  pwd->username = uv__malloc(name_size + homedir_size + shell_size + gecos_size);
```

Comment:
Also, I wouldn't want to vouch that `pw_gecos` is never NULL so maybe add NULL checks.